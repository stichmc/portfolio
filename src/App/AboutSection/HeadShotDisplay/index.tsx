import { div as MotionDiv } from "motion/react-client";

export interface HeadShotDisplayProps {
  headShotImage: string;
}

const HeadShotDisplay = ({ headShotImage }: HeadShotDisplayProps) => {
  return (
    <div className="relative w-[200px] h-[200px] sm:w-[250px] sm:h-[250px] md:w-[300px] md:h-[300px] lg:w-[350px] lg:h-[350px] xl:w-[400px] xl:h-[400px]">
      <div className="absolute inset-0 z-0 flex items-center justify-center">
        <MotionDiv
          animate={{
            scale: [1, 2, 2, 1, 1],
            rotate: [0, 0, 180, 180, 0],
            borderRadius: ["0%", "0%", "50%", "50%", "0%"],
          }}
          transition={{
            duration: 8,
            ease: "easeInOut",
            times: [0, 0.2, 0.5, 0.8, 1],
            repeat: Infinity,
            repeatDelay: 2,
            delay: 2,
          }}
          className="bg-card-foreground w-[100px] h-[100px] sm:w-[125px] sm:h-[125px] md:w-[150px] md:h-[150px] lg:w-[175px] lg:h-[175px] xl:w-[200px] xl:h-[200px]"
        />
      </div>
      <div
        className="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4"
        // Ensure Safari renders this animation correctly
        style={{ transform: "translateZ(0)" }}
      >
        <img className="w-full h-full object-cover" src={headShotImage} alt="michael-headshot" />
      </div>
    </div>
  );
};

export default HeadShotDisplay;
