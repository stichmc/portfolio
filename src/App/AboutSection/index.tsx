import { Code, Github, Linkedin, Mails } from "lucide-react";
import { div as MotionDiv } from "motion/react-client";

import FadeUp from "@/components/FadeUp";
import { Button } from "@/components-shadcn/button";
import { ButtonGroup } from "@/components-shadcn/button-group";

import HeadShotDisplay from "./HeadShotDisplay";

export interface AboutSectionProps {
  headShotImage: string;
}

const AboutSection = ({ headShotImage }: AboutSectionProps) => {
  return (
    <div className="flex flex-col-reverse lg:grid lg:grid-cols-2 gap-24 items-center justify-center h-[80vh] max-w-[1500px] sm:p-12 p-4">
      <div className="flex flex-col items-center gap-4 justify-center text-center">
        <FadeUp duration={0.5}>
          <h1 className="scroll-m-20 text-center text-4xl font-extrabold text-balance">
            <MotionDiv
              style={{ display: "inline-block", paddingRight: 12 }}
              animate={{
                rotate: [0, 14, -8, 14, -4, 10, 0],
              }}
              transition={{
                duration: 2.5,
                ease: "easeInOut",
                repeat: Infinity,
              }}
            >
              ðŸ‘‹
            </MotionDiv>
            Hi, I'm Michael Stich
          </h1>
        </FadeUp>

        <FadeUp delay={0.75} duration={0.5} className="max-w-[500px]">
          <h2 className="scroll-m-20 text-2xl font-semibold tracking-tight first:mt-0">
            A software engineer with 3 years of professional work experience.
          </h2>
        </FadeUp>

        <FadeUp delay={1.5} duration={0.5} className="max-w-[550px]">
          <ButtonGroup>
            <Button
              variant="outline"
              className="flex items-center justify-center gap-2 sm:text-lg"
              onClick={() => window.open("mailto:mcstich@outlook.com")}
            >
              <Mails /> Email
            </Button>
            <Button
              variant="outline"
              className="flex items-center justify-center gap-2 sm:text-lg"
              onClick={() => window.open("https://linkedin.com/in/mcstich/", "_blank")}
            >
              <Linkedin /> LinkedIn
            </Button>
            <Button
              variant="outline"
              className="flex items-center justify-center gap-2 sm:text-lg"
              onClick={() => window.open("https://github.com/stichmc/", "_blank")}
            >
              <Github /> Github
            </Button>

            <Button
              variant="outline"
              className="flex items-center justify-center gap-2 sm:text-lg"
              onClick={() => window.open("https://github.com/stichmc/portfolio/", "_blank")}
            >
              <Code /> Source Code
            </Button>
          </ButtonGroup>
        </FadeUp>

        <FadeUp delay={1.5} duration={0.5} className="max-w-[550px]">
          <h3 className="leading-6">
            I build and scale applications for large user bases across the entire stack: frontend, backend, and cloud.
            I'm very passionate about solving complex problems and working with extraordinary people.
          </h3>
        </FadeUp>
      </div>

      <div className="flex items-center justify-center">
        <HeadShotDisplay headShotImage={headShotImage} />
      </div>
    </div>
  );
};

export default AboutSection;
