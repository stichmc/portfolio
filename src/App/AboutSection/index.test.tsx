import { expect, test, beforeEach, vi } from "vitest";
import { render } from "@testing-library/react";
import AboutSection, { AboutSectionProps } from ".";

let props: AboutSectionProps;

beforeEach(() => {
  props = {
    headShotImage: "link-to-image",
  };

  const mockIntersectionObserver = vi.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null,
  });
  window.IntersectionObserver = mockIntersectionObserver;
});

test("AboutSection", () => {
  const rendering = render(<AboutSection {...props} />);
  expect(rendering).toMatchSnapshot();
});
