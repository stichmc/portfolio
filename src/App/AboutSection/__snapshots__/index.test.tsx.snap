// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutSection 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col-reverse lg:grid lg:grid-cols-2 gap-24 items-center justify-center h-[80vh] max-w-[1500px] sm:p-12 p-4"
      >
        <div
          class="flex flex-col items-center gap-4 justify-center text-center"
        >
          <div
            style="opacity: 0; transform: translateY(15px);"
          >
            <div
              class=""
            >
              <h1
                class="scroll-m-20 text-center text-4xl font-extrabold text-balance"
              >
                <div
                  style="display: inline-block; padding-right: 12px;"
                >
                  ðŸ‘‹
                </div>
                Hi, I'm Michael Stich
              </h1>
            </div>
          </div>
          <div
            style="opacity: 0; transform: translateY(15px);"
          >
            <div
              class="max-w-[500px]"
            >
              <h2
                class="scroll-m-20 text-2xl font-semibold tracking-tight first:mt-0"
              >
                A software engineer with 3 years of professional work experience.
              </h2>
            </div>
          </div>
          <div
            style="opacity: 0; transform: translateY(15px);"
          >
            <div
              class="max-w-[550px]"
            >
              <div
                class="flex w-fit items-stretch has-[>[data-slot=button-group]]:gap-2 [&>*]:focus-visible:relative [&>*]:focus-visible:z-10 has-[select[aria-hidden=true]:last-child]:[&>[data-slot=select-trigger]:last-of-type]:rounded-r-md [&>[data-slot=select-trigger]:not([class*='w-'])]:w-fit [&>input]:flex-1 [&>*:not(:first-child)]:rounded-l-none [&>*:not(:first-child)]:border-l-0 [&>*:not(:last-child)]:rounded-r-none"
                data-slot="button-group"
                role="group"
              >
                <button
                  class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
                >
                  <svg
                    class="lucide lucide-mails "
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      width="16"
                      x="6"
                      y="4"
                    />
                    <path
                      d="m22 7-7.1 3.78c-.57.3-1.23.3-1.8 0L6 7"
                    />
                    <path
                      d="M2 8v11c0 1.1.9 2 2 2h14"
                    />
                  </svg>
                   Email
                </button>
                <button
                  class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
                >
                  <svg
                    class="lucide lucide-linkedin "
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                    />
                    <rect
                      height="12"
                      width="4"
                      x="2"
                      y="9"
                    />
                    <circle
                      cx="4"
                      cy="4"
                      r="2"
                    />
                  </svg>
                   LinkedIn
                </button>
                <button
                  class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
                >
                  <svg
                    class="lucide lucide-github "
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                    />
                    <path
                      d="M9 18c-4.51 2-5-2-7-2"
                    />
                  </svg>
                   Github
                </button>
                <button
                  class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
                >
                  <svg
                    class="lucide lucide-code "
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="16 18 22 12 16 6"
                    />
                    <polyline
                      points="8 6 2 12 8 18"
                    />
                  </svg>
                   Source Code
                </button>
              </div>
            </div>
          </div>
          <div
            style="opacity: 0; transform: translateY(15px);"
          >
            <div
              class="max-w-[550px]"
            >
              <h3
                class="leading-6"
              >
                I build and scale applications for large user bases across the entire stack: frontend, backend, and cloud. I'm very passionate about solving complex problems and working with extraordinary people.
              </h3>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center"
        >
          <div
            class="relative w-[400px] h-[400px]"
          >
            <div
              class="absolute inset-0 z-0 flex items-center justify-center"
            >
              <div
                class="bg-card-foreground w-[200px] h-[200px]"
              />
            </div>
            <div
              class="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4"
            >
              <img
                alt="michael-headshot"
                class="w-full h-full object-cover"
                src="link-to-image"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col-reverse lg:grid lg:grid-cols-2 gap-24 items-center justify-center h-[80vh] max-w-[1500px] sm:p-12 p-4"
    >
      <div
        class="flex flex-col items-center gap-4 justify-center text-center"
      >
        <div
          style="opacity: 0; transform: translateY(15px);"
        >
          <div
            class=""
          >
            <h1
              class="scroll-m-20 text-center text-4xl font-extrabold text-balance"
            >
              <div
                style="display: inline-block; padding-right: 12px;"
              >
                ðŸ‘‹
              </div>
              Hi, I'm Michael Stich
            </h1>
          </div>
        </div>
        <div
          style="opacity: 0; transform: translateY(15px);"
        >
          <div
            class="max-w-[500px]"
          >
            <h2
              class="scroll-m-20 text-2xl font-semibold tracking-tight first:mt-0"
            >
              A software engineer with 3 years of professional work experience.
            </h2>
          </div>
        </div>
        <div
          style="opacity: 0; transform: translateY(15px);"
        >
          <div
            class="max-w-[550px]"
          >
            <div
              class="flex w-fit items-stretch has-[>[data-slot=button-group]]:gap-2 [&>*]:focus-visible:relative [&>*]:focus-visible:z-10 has-[select[aria-hidden=true]:last-child]:[&>[data-slot=select-trigger]:last-of-type]:rounded-r-md [&>[data-slot=select-trigger]:not([class*='w-'])]:w-fit [&>input]:flex-1 [&>*:not(:first-child)]:rounded-l-none [&>*:not(:first-child)]:border-l-0 [&>*:not(:last-child)]:rounded-r-none"
              data-slot="button-group"
              role="group"
            >
              <button
                class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
              >
                <svg
                  class="lucide lucide-mails "
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="13"
                    rx="2"
                    width="16"
                    x="6"
                    y="4"
                  />
                  <path
                    d="m22 7-7.1 3.78c-.57.3-1.23.3-1.8 0L6 7"
                  />
                  <path
                    d="M2 8v11c0 1.1.9 2 2 2h14"
                  />
                </svg>
                 Email
              </button>
              <button
                class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
              >
                <svg
                  class="lucide lucide-linkedin "
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                  />
                  <rect
                    height="12"
                    width="4"
                    x="2"
                    y="9"
                  />
                  <circle
                    cx="4"
                    cy="4"
                    r="2"
                  />
                </svg>
                 LinkedIn
              </button>
              <button
                class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
              >
                <svg
                  class="lucide lucide-github "
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                  />
                  <path
                    d="M9 18c-4.51 2-5-2-7-2"
                  />
                </svg>
                 Github
              </button>
              <button
                class="whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 flex items-center justify-center gap-2 sm:text-lg"
              >
                <svg
                  class="lucide lucide-code "
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
                 Source Code
              </button>
            </div>
          </div>
        </div>
        <div
          style="opacity: 0; transform: translateY(15px);"
        >
          <div
            class="max-w-[550px]"
          >
            <h3
              class="leading-6"
            >
              I build and scale applications for large user bases across the entire stack: frontend, backend, and cloud. I'm very passionate about solving complex problems and working with extraordinary people.
            </h3>
          </div>
        </div>
      </div>
      <div
        class="flex items-center justify-center"
      >
        <div
          class="relative w-[400px] h-[400px]"
        >
          <div
            class="absolute inset-0 z-0 flex items-center justify-center"
          >
            <div
              class="bg-card-foreground w-[200px] h-[200px]"
            />
          </div>
          <div
            class="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4"
          >
            <img
              alt="michael-headshot"
              class="w-full h-full object-cover"
              src="link-to-image"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
